### PR TITLE
Add sheets2* mashup samples.

### PR DESCRIPTION
**Pull request by erickoledadevrel**
A first round of mashup samples, using Sheets as the dat a source.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,1171 @@
+{
```
Comment (grant) [inline package-lock.json]: Can we add this in a different PR?

File: mashups/sheets2contacts.gs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * columns "First Name", "Last Name", and "Email" in that order.
+ */
+function createContactsFromSpreadsheet() {
+  // Open the spreadsheet and get the data.
+  var ss = SpreadsheetApp.openByUrl('ENTER SPREADSHEET URL HERE');
+  var sheet = ss.getSheets()[0];
+  var data = sheet.getDataRange().getValues();
+
+  // Remove any frozen rows from the data, since they contain headers.
+  for (var i = 0; i < sheet.getFrozenRows(); i++) {
```
Comment (grant) [inline mashups/sheets2contacts.gs]: If we're using `forEach`, then we could use it consistently:
```js
sheet.getFrozenRows().forEach(data.shift);
```
this would replace the for loop.

File: mashups/sheets2docs.gs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * addresses separated by a comma.
+ */
+function createDocsFromSpreadsheet() {
+  // Open the spreadsheet and get the data.
+  var ss = SpreadsheetApp.openByUrl('ENTER SPREADSHEET URL HERE');
+  var sheet = ss.getSheets()[0];
+  var data = sheet.getDataRange().getValues();
+
+  // Remove any frozen rows from the data, since they contain headers.
+  for (var i = 0; i < sheet.getFrozenRows(); i++) {
```
Comment (grant) [inline mashups/sheets2docs.gs]: Ditto above (`forEach`)

File: mashups/sheets2docs.gs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    var content = row[1];
+    var emails = row[2];
+
+    // Split the emails into an array and remove extra whitespace.
+    emails = emails.split(',').map(function(email) {
+      return email.trim();
+    });
+
+    // Create the document, append the content, and share it out.
+    var doc = DocumentApp.create(title);
```
Comment (grant) [inline mashups/sheets2docs.gs]: You could probably chain as such:
```js
DocumentApp.create(title)
    .addEditors(emails)
    .getBody().appendParagraph(content);
```
The order in which you add editors and modify the body shouldn't matter.

File: mashups/sheets2maps.gs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * This script must be attached to the spreadsheet (created in Google Sheets
+ * under "Tools > Script editor").
+ *
+ * @param {String} address The address to lookup.
+ * @return {Number} The county (or equivalent) the address is within.
+ * @customFunction
+ */
+function COUNTY(address) {
+  var response = Maps.newGeocoder().geocode(address);
```
Comment (grant) [inline mashups/sheets2maps.gs]: We don't use `response` without `results`, so we can use `results` instead:

```js
var results = Maps.newGeocoder().geocode(address).results;
if (!results || results.length === 0) {
  ...
}
```

File: mashups/sheets2maps.gs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  if (!response.results || response.results.length === 0) {
+    throw new Error('Unknown address');
+  }
+  var result = response.results[0];
```
Comment (grant) [inline mashups/sheets2maps.gs]: I think `firstResult` is a better name than `result`.
However, it's only used once, so just inline it like the other case (`counties[0].long_name`)

File: mashups/sheets2translate.gs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+ * This script must be attached to the spreadsheet (created in Google Sheets
+ * under "Tools > Script editor").
+ *
+ * @param {String} text The text to translate.
+ * @param {String} fromLanguageCode The language code of the source text. Use
+ *     an empty string to have the auto-detect the language.
+ * @param {String} toLanguageCode The language code of the language to translate
+ *     the text to.
+ * @return {String} The translated text.
+ * @customFunction
```
Comment (grant) [inline mashups/sheets2translate.gs]: Can you mention somewhere that this custom function is also built into Google Sheets?
`=GOOGLETRANSLATE`
https://support.google.com/docs/answer/3093331?hl=en

File: mashups/sheets2translate.gs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+ */
+function TRANSLATE(text, fromLanguageCode, toLanguageCode) {
```
Comment (grant) [inline mashups/sheets2translate.gs]: Can we use `sourceLanguage` and `targetLanguage`?
- `from` and `to` aren't great identifiers.
- The `Code` part is not necessary and can be confusing. Can we not include it to be consistent with the [docs](https://developers.google.com/apps-script/reference/language/language-app#translate(String,String,String))? If we want to be more specific, can we mention the ISO 639-1 code?

File: mashups/sheets2contacts.gs (hunk 2 above)
Comment (erickoledadevrel) [reply mashups/sheets2contacts.gs]: getFrozenRows() returns a number, not an array, so forEach won't work here.

File: mashups/sheets2docs.gs (hunk 3 above)
Comment (erickoledadevrel) [reply mashups/sheets2docs.gs]: Ditto

File: mashups/sheets2docs.gs (hunk 4 above)
Comment (erickoledadevrel) [reply mashups/sheets2docs.gs]: I'd really like to stick with the original order, since I think it matches the mental model a new developer would have. This order would also make it harder to extend, since the resulting object would be the Body and there would be no handle on the doc. Keeping it as-is in this case.

File: mashups/sheets2maps.gs (hunk 6 above)
Comment (erickoledadevrel) [reply mashups/sheets2maps.gs]: I had it in-lined originally, but the line became too long and a little hard to parse. Changing to `firstResult`.

File: mashups/sheets2translate.gs (hunk 7 above)
Comment (erickoledadevrel) [reply mashups/sheets2translate.gs]: Whoops. I had checked for "TRANSLATE" but forgot about "GOOGLETRANSLATE". I'll change this sample to do something different.

File: mashups/sheets2contacts.gs (hunk 2 above)
Comment (grant) [reply mashups/sheets2contacts.gs]: OK.
Can you use `data.splice` rather than `data.shift` n times?
This is like pushing to an array n times rather than concatenating once.

File: package-lock.json (hunk 1 above)
Comment (erickoledadevrel) [reply package-lock.json]: Seems like more work now to back it out. When should we be updating this then?

File: mashups/sheets2contacts.gs (hunk 2 above)
Comment (erickoledadevrel) [reply mashups/sheets2contacts.gs]: Good call! I always forget about splice.

File: package-lock.json (hunk 1 above)
Comment (erickoledadevrel) [reply package-lock.json]: Think I figured out how to remove this.

**Comment by grant** [comment]
This PR was approved in the previous review.
Feel free to merge.
